server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java

```diff
@@ -80,7 +81,7 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
         @Override
         public void enQueue(Mail mail) throws MailQueueException {
             try {
-                mailItems.addFirst(new MemoryMailQueueItem(cloneMail(mail)));
+                mailItems.addLast(new MemoryMailQueueItem(cloneMail(mail)));
```

Comment:
does that mean our contract are not good enough ?

Reply:
No, I just switch the order we use in the queue to match the semantic `getLast` and avoid a getLast calling getFirst what would have been extremly *confusing*.